pkg/controller/operator.go

```diff
@@ -77,6 +79,27 @@ func (s *Support) Run(controller *controllercmd.ControllerContext) error {
 	if len(s.Impersonate) > 0 {
 		gatherKubeConfig.Impersonate.UserName = s.Impersonate
 	}
+
+	// the metrics client will connect to prometheus and scrape a small set of metrics
+	// TODO: the oauth-proxy and delegating authorizer do not support Impersonate-User,
+	//   so we do not impersonate gather
+	metricsGatherKubeConfig := rest.CopyConfig(controller.KubeConfig)
+	metricsGatherKubeConfig.CAFile = "/var/run/secrets/kubernetes.io/serviceaccount/service-ca.crt"
+	metricsGatherKubeConfig.NegotiatedSerializer = scheme.Codecs
+	metricsGatherKubeConfig.GroupVersion = &schema.GroupVersion{}
+	metricsGatherKubeConfig.APIPath = "/"
+	metricsGatherKubeConfig.Host = "https://prometheus-k8s.openshift-monitoring.svc:9091"
+
+	// If we fail, it's likely due to the service CA not existing yet. Warn and continue,
+	// and when the service-ca is loaded we will be restarted.
+	var metricsClient rest.Interface
+	metricsRESTClient, err := rest.RESTClientFor(metricsGatherKubeConfig)
+	if err != nil {
+		klog.Warningf("Unable to load metrics client, no metrics will be collected: %v", err)
```

Comment:
should this be degraded condition?

Reply:
I'd probably say no, because if service-ca.crt is broken everything else should break, and we don't need prom metrics.  I may have metrics gracefully handle down prometheus (was looking at the errors we get during startup) simply to avoid confusing the issue.  Insights operator isn't responsible for reporting health of other components today.

Reply:
@smarterclayton it is not other component health, I think if insights operator can't get data it should write to report, it should be communicated and visible to user/admin (not just logged...) maybe warning event? anyway, can be a follow up

Reply:
I made even metrics retrieval errors survivable, and instead have them record something in the insights data (comments are allowed in the file format).